megamek/src/megamek/server/commands/ConvertMMSCommand.java

```diff
@@ -0,0 +1,206 @@
+            
+            // Factions section.
+            String[] factions = {};
+            int[] playerPos = {};
+            writer.write(PARAM_FACTIONS + SEPARATOR_PROPERTY);
+            for (Enumeration<IPlayer> players = server.getGame().getPlayers(); players.hasMoreElements();) {
+                 if (factions.length != 0) {
+                     writer.append(SEPARATOR_COMMA);
+                 }
+                 IPlayer p = players.nextElement();
+                 writer.append(p.getName());
+                 factions = Arrays.copyOf(factions, factions.length + 1);
+                 factions[factions.length - 1] = p.getName();
+                playerPos = Arrays.copyOf(playerPos, playerPos.length + 1);
+                playerPos[playerPos.length - 1] = p.getStartingPos();
+            }
+            writer.newLine();
+            
+            // Starting Locations
+            for (int i = 0; i < factions.length; i++) {
+                writer.write(PARAM_LOCATION + SEPARATOR_UNDERSCORE + factions[i] + SEPARATOR_PROPERTY + 
+                             IStartingPositions.START_LOCATION_NAMES[playerPos[i]]);
+                writer.newLine();
+            }
+            
+            // Units output
+            
+            for (Enumeration<IPlayer> players = server.getGame().getPlayers(); players.hasMoreElements();) {
+                IPlayer p = players.nextElement();
+                int numUnits = server.getGame().getEntitiesOwnedBy(p);
+                ArrayList<Entity> PlayerUnits = server.getGame().getPlayerEntities(p, true);
+                
+                for (int counter = 0; counter < PlayerUnits.size(); counter++) {
+                    writer.write("Unit" + SEPARATOR_UNDERSCORE + p.getName() +
+                                 SEPARATOR_UNDERSCORE + (counter+1) + SEPARATOR_PROPERTY +
+                                 PlayerUnits.get(counter).getShortName() + SEPARATOR_COMMA +
+                                 PlayerUnits.get(counter).getCrew().getName() + SEPARATOR_COMMA +
+                                 PlayerUnits.get(counter).getCrew().getGunnery() + SEPARATOR_COMMA +
+                                 PlayerUnits.get(counter).getCrew().getPiloting());
+                    writer.newLine();
+                }
+                // Need to add lines to look up the mech position, and add it in terms of distance from NW corner (N,0,0 is NW corner)
+                // Values are X,Y
+                
+                // need to look at damage. For mechs this is DamageSpecific=N<X>:<remaining armor>
+                // eg: N0:5,N1:14,R1:3,N2:11,R2:3,N3:11,R3:3,N4:9,N5:9,N6:12,N7:12
+                // These are: N0=Head, N1=CT, R1=CTR, N2=RT,R2=RTR,N3=LT,R3=RTR,N4=RA,N5=LA,N6=RL,N7=RL
+                
+                
+                // For random damage, use Unit_<player>_<unitNum>_Damage=<amount>
+            }
+            
+            writer.close();
+        } catch (IOException e) {
```

Comment:
```suggestion
        } catch (IOException ignored) {
```

Reply:
Alternatively, I'd add a logging statement to inform us as to why the write failed.